### PR TITLE
Add polling support to caching DAO

### PR DESCRIPTION
**Pull request by lchanmann**
## Ref
- https://nanopay.atlassian.net/browse/NP-5405

## Changes
- Add polling support to caching DAO
- Add polling cache configs to easy DAO

### CODE REVIEW

File: src/foam/dao/CachingDAO.js [hunk 1]
```diff
@@ -135,11 +154,26 @@ foam.CLASS({
...
+
+    function poll() {
+      var self = this;
+
+      self.delegate
+        .orderBy(self.pollingComparator).limit(1)
+        .select().then(function(data) {
+          if ( data.array.length === 1 ) {
+            self.src
+              .where(self.GT(self.src.of.ID, data.array[0].id))
```
Comment (jlhughes) [inline src/foam/dao/CachingDAO.js]: what is this?

Comment (lchanmann) [reply src/foam/dao/CachingDAO.js]: Refactored to polling property https://github.com/kgrgreer/foam3/pull/881/commits/16a898c91654fca6478bfc928148bc3c052f9d94.

**Review by sarthak-marwaha — COMMENTED** [review]
LGTM

File: src/foam/dao/CachingDAO.js [hunk 2]
```diff
@@ -87,6 +93,15 @@ foam.CLASS({
           promise: cacheFilled
         });
       }
+    },
+    {
+      class: 'Int',
+      name: 'pollingFrequency'
```
Comment (kgrgreer) [inline src/foam/dao/CachingDAO.js]: frequency is times/unit-of-time, like 3hz, 10 times / minute, etc. but this is a duration/delay/interval, so maybe pollingInterval would be better. And add:
units: 'ms',

File: src/foam/dao/CachingDAO.js [hunk 3]
```diff
@@ -135,10 +154,26 @@ foam.CLASS({
...
         delete this.private_['delegate'];
       } else if ( this.PurgeRecordCmd.isInstance(obj) ) {
+        // REVIEW: this.cache is a dao not object, need to call dao.remove(obj)?
         delete this.cache[obj.id];
       } else {
         this.SUPER(x, obj);
       }
+    },
+
+    function poll() {
```
Comment (kgrgreer) [inline src/foam/dao/CachingDAO.js]: This should be in listeners: and then you don't have to bind() it.

File: src/foam/nanos/u2/navigation/NotificationMenuItem.js [hunk 4]
```diff
@@ -94,9 +93,8 @@ foam.CLASS({
 
   methods: [
     function render() {
-      this.notificationDAO.on.sub(this.onDAOUpdate);
-      this.user$.dot('id').sub(this.onDAOUpdate);
-      this.group$.dot('id').sub(this.onDAOUpdate);
+      this.myNotificationDAO.on.sub(this.onDAOUpdate);
```
Comment (kgrgreer) [inline src/foam/nanos/u2/navigation/NotificationMenuItem.js]: Connect to detach when Element removed:
this.onDetach(this.myNotificationDAO.on.sub(this.onDAOUpdate));

Same for the next line as well.

**Pull request merged**